src/protobuf-net.Grpc.AspNetCore/ServicesExtensions.cs

```diff
@@ -17,7 +18,7 @@ public static class ServicesExtensions
         /// <summary>
         /// Registers a provider that can recognize and handle code-first services
         /// </summary>
-        public static void AddCodeFirstGrpc(this IServiceCollection services) => AddCodeFirstGrpc(services, null);
+        public static IGrpcServerBuilder AddCodeFirstGrpc(this IServiceCollection services) => AddCodeFirstGrpc(services, null);
```

Comment:
note that this represents a hard break for semver purposes; I wonder whether it is possible to do instead:

``` c#
// note no "this", so it is no longer an extension method and will not be selected by new builds
public static void AddCodeFirstGrpc(IServiceCollection services) => AddCodeFirstGrpc(services, null);
public static IGrpcServerBuilder AddCodeFirstGrpc(this IServiceCollection services, Action<BinderOptions> binderOptions = null)
{...}
```

This avoids a hard break but gives you the same results.

Reply:
What is the preference for when we also need the other override for `GrpcServiceOptions`? It's why I added as an extension instead of an override to `AddCodeFirstGrpc`
I could either add as an extra field to the `BinderOptions`(name change to `CodeFirstGrpcConfiguration`) or would you prefer dual options, Not sure I like adding it as an extra field as it would be added as an `Action<GrpcServiceOptions>` property as the underlying Configure is passed the action to lazy evaluate, which is slightly different to the eager evaluation of the `CodeFirstGrpcConfiguration`
```
//Existing + suggested override
public static void AddCodeFirstGrpc(IServiceCollection services) => AddCodeFirstGrpc(services, null);
public static IGrpcServerBuilder AddCodeFirstGrpc(this IServiceCollection services, Action<GrpcServiceOptions>? configureOptions)
public static IGrpcServerBuilder AddCodeFirstGrpc(this IServiceCollection services, Action<CodeFirstGrpcConfiguration>? grpcConfiguration = null)
// Extra
public static IGrpcServerBuilder AddCodeFirstGrpc(this IServiceCollection services, Action<GrpcServiceOptions>? configureOptions, Action<CodeFirstGrpcConfiguration>? grpcConfiguration)

// Or just switch to non action/option/config style
public static void AddCodeFirstGrpc(IServiceCollection services) => AddCodeFirstGrpc(services, null);
public static IGrpcServerBuilder AddCodeFirstGrpc(this IServiceCollection services, Action<GrpcServiceOptions>? configureOptions)
public static IGrpcServerBuilder AddCodeFirstGrpc(this IServiceCollection services, BinderConfiguration? binderConfiguration = null)
// Add Create override to to allow not setting a marshaller.
services.AddCodeFirstGrpc(BinderConfiguration.Create(new MyServiceBinder));
```

Reply:
Think my preference would be just to get rid of my `BinderOptions`/`CodeFirstGrpcConfiguration`
and go with an extra override to `Create` to remove the null marshaller requirement.